core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcConvention.java

```diff
@@ -49,12 +51,23 @@
 
   public final SqlDialect dialect;
   public final Expression expression;
+  private final String signature;
 
   public JdbcConvention(SqlDialect dialect, Expression expression,
       String name) {
     super("JDBC." + name, JdbcRel.class);
     this.dialect = dialect;
     this.expression = expression;
+    this.signature = normalizeName();
+  }
+
+  private String normalizeName() {
+    String id = UUID.randomUUID().toString();
+    return getName().replaceAll("[^-A-Za-z0-9_.():]", ".") + "(" + id + ")";
+  }
```

Comment:
Why we use a uuid a part of the name ? So you mean for same name we can have different signature ?

Reply:
I think it is done to avoid collisions that may occur after the normalization.